### PR TITLE
start-qemu.sh: add support for TD quote generation

### PR DESCRIPTION
**Pull request by jurobystricky**
There are two ways to run quote generation in TD:

1. Get quote with vsock
2. Get quote with tdvmcall

This patch allows user to specify quote generation type via an additional
optional argument 'q':

 -q vsock
 -q tdvmcall

When requesting a quote via 'vsock', 'vsock' is enabled and the kernel command
line will contain 'tdx_disable_filter'. (User is still responsible for the creation
of the required file /etc/tdx-attest.conf in TD VM needed in this case.)

When requesting a quote via 'tdvmcall', the qemu command line
  '-object tdx-guest,id=tdx'
is modified to
  '-object tdx-guest,id=tdx,quote-generation-service=vsock:2:4050'

Examples:

  ./start-qemu.sh -i td-guest-rhel-8.5.qcow2 -k vmlinuz-rhel-8.5 -q vsock
  ./start-qemu.sh -i td-guest-rhel-8.5.qcow2 -k vmlinuz-rhel-8.5 -q tdvmcall

Signed-off-by: Juro Bystricky <juro.bystricky@intel.com>

### CODE REVIEW

File: start-qemu.sh [hunk 1]
```diff
@@ -159,6 +162,19 @@ process_args() {
         OVMF_CODE="/usr/share/qemu/OVMF_CODE.debug.fd"
     fi
 
+    if [[ -n ${QUOTE_TYPE} ]]; then
```
Comment (jialeif) [inline start-qemu.sh:165]: could it be as simple as:
            if [[ ${QUOTE_TYPE} == "vsock" ]]; then
                ...
            fi
You can add one or more reviewers

Comment (kenplusplus) [reply start-qemu.sh:165]: @jialeif  I think Juro switch case is simpler and cleaner in logic than explicity if [[ ${QUOTE_TYPE} == "vsock" ]].

File: start-qemu.sh [hunk 2]
```diff
@@ -159,6 +162,19 @@ process_args() {
...
+        case ${QUOTE_TYPE} in
+            "tdvmcall") ;;
+            "vsock")
+                KERNEL_CMD_TD+=" tdx_disable_filter"
```
Comment (kenplusplus) [inline start-qemu.sh:169]: @jurobystricky  I think we may add comment like:
vsock drive is filtered in TD guest kernel by default due to security consideration, so need addtional parameter "tdx_disable_filter". But "tdx_disable_filter" is not recommend.

**Review by kenplusplus — APPROVED** [review]
Nice fixing for attestation feature via QEMU. This patch looks good to me. Beyond start-qemu.sh, we may also need provide attestation template for libvirt like https://github.com/intel/tdx-tools/blob/main/doc/tdx_libvirt_grub.xml.template, @jurobystricky can you continue to help?

**Review by clsulliv — APPROVED** [review]

**Comment by clsulliv** [comment]
We will need to document the host and guest setup when the attestation components are available, but this looks good for now.

**Pull request merged**